File: Data Structures & Algorithms/anagram-groups/submission-14.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        frequency = defaultdict(list)
        for string in strs:
            count = [0]*26
            for c in string:
                count[ord(c) - ord('a')] += 1
            frequency[tuple(count)].append(string)
        return list(frequency.values())

        

File: Data Structures & Algorithms/is-anagram/submission-14.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s == null || t == null) {
            return false;
        }
        if (s.length() != t.length()) {
            return false;
        }
        final int len = s.length();
        final int[] count = new int[26];
        for (int i = 0; i < len; i++) {
            count[s.charAt(i) - 'a']++;
            count[t.charAt(i) - 'a']--;
        }
        for (int val : count) {
            if (val != 0) {
                return false;
            }
        }
        return true;
    }
}
